File: database/database-api-model/src/main/java/io/ehdev/conrad/database/model/project/DefaultApiRepoModel.java
package io.ehdev.conrad.database.model.project;

public class DefaultApiRepoModel implements ApiFullRepoModel {

    private final String url;
    private final String projectName;
    private final String repoName;

    public DefaultApiRepoModel(String projectName, String repoName, String url) {
        this.url = url;
        this.projectName = projectName;
        this.repoName = repoName;
    }

    public DefaultApiRepoModel(String projectName, String repoName) {
        this(projectName, repoName, null);
    }

    @Override
    public String getUrl() {
        return url;
    }

    @Override
    public String getProjectName() {
        return projectName;
    }

    @Override
    public String getRepoName() {
        return repoName;
    }

    @Override
    public String getMergedName() {
        return projectName + "/" + repoName;
    }
}

File: database/database-api-model/src/main/kotlin/io/ehdev/conrad/database/model/project/DefaultApiRepoModel.kt
package io.ehdev.conrad.database.model.project

class DefaultApiRepoModel(
    override val projectName: String,
    override val repoName: String,
    override val url: String?
) : ApiFullRepoModel  {
    constructor(projectName: String, repoName: String) : this(projectName, repoName, null)
}
